lib/ansible/plugins/action/template.py

```diff
@@ -92,7 +94,11 @@ def run(self, tmp=None, task_vars=None):
         b_source = to_bytes(source)
         try:
             with open(b_source, 'r') as f:
-                template_data = to_text(f.read())
+                template_data = None
+                if decode is not None:
+                    template_data = to_text(f.read(), encoding=decode)
+                else:
+                    template_data = to_text(f.read())
```

Comment:
I notice that the use of to_text() here is making undecodable characters be replaced.  Is that what we want?  Or do we want to make it an error (or at least, a warning)?  (Implementing a warning is done by first using errors='strict'.  If an exception is thrown, print the warning and do it a second time with errors='surrogate_or_replace').